pycc/ccwfn.py

```diff
@@ -364,6 +371,22 @@ def r_T2(self, o, v, F, ERI, L, t1, t2, Fae, Fme, Fmi, Wmnij, Wmbej, Wmbje, Zmbi
             r_T2 = r_T2 + contract('imae,mbej->ijab', (t2 - t2.swapaxes(2,3)), Wmbej)
             r_T2 = r_T2 + contract('imae,mbej->ijab', t2, (Wmbej + Wmbje.swapaxes(2,3)))
             r_T2 = r_T2 + contract('mjae,mbie->ijab', t2, Wmbje)
+        elif self.model == 'CC2':
```

Comment:
I'm not sure whether an entirely new code block for the CC2 case is what we want because there's so much redundancy with CCSD.  Do you think the code is more readable this way?

Please remove any `fact1=1.0` arguments since that's the default.

Also, we should be sure to skip construction of any intermediates that aren't actually needed by CC2, especially Wmbej and Wbmje, which are rather expensive to build.  (See `residuals()`.)

Reply:
what do you think is the cleaner option: `if` statements in `residuals()`, or 'if self.model == "CC2": return` in Wmbej and Wbmje (as in Zmbij for the CCD case)?

Reply:
I think the latter is cleaner.

Reply:
I did put an entirely new code block for CC2 for the readability, but I can change that if you want.

Reply:
No, I'm genuinely not sure which is the better option here.  Let's leave it as it is for now.  It doesn't have to be decided in this PR.